packages/shell-api/src/shard.ts

```diff
@@ -424,4 +426,29 @@ export default class Shard extends ShellApiWithMongoClass {
     }
     return this.stopBalancer();
   }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['6.1.0', ServerVersions.latest])
+  @returnType('AggregationCursor')
+  async getShardedDataDistribution(options = {}): Promise<AggregationCursor> {
+    this._emitShardApiCall('getShardedDataDistribution', {});
+
+    const cursor = await this._database.aggregate([{ $shardedDataDistribution: options }]);
```

Comment:
Let’s make this 
```suggestion
    const cursor = await this._database.getSiblingDB('admin').aggregate([{ $shardedDataDistribution: options }]);
```

if it’s important to run it against the admin database (that matches the behavior of the methods above which use `_runAdminCommand` instead of `_runCommand ` to transparently select the right db)

Reply:
Nice 😄